src/library_pthread.js

```diff
@@ -402,8 +402,24 @@ var LibraryPThread = {
 #if PTHREADS_DEBUG
         out('Allocating a new web worker from ' + new URL('{{{ PTHREAD_WORKER_FILE }}}', import.meta.url));
 #endif
-        // Use bundler-friendly `new Worker(new URL(..., import.meta.url))` pattern; works in browsers too.
+#if TRUSTED_TYPES
+        // Use Trusted Types compatible wrappers.
+        if (typeof trustedTypes !== 'undefined' && trustedTypes.createPolicy) {
+          var p = trustedTypes.createPolicy(
+            'emscripten#workerPolicy1',
+            {
+              createScriptURL: function(ignored) {
+                return new URL('{{{ PTHREAD_WORKER_FILE }}}', import.meta.url);
+              }
+            }
+          );
+          PThread.unusedWorkers.push(new Worker(p.createScriptURL('ignored')));
+        } else {
+          PThread.unusedWorkers.push(new Worker(new URL('{{{ PTHREAD_WORKER_FILE }}}', import.meta.url)));
```

Comment:
It would be good to avoid this duplication, as this line is identical to the new 421 just below it. Getting the ifdefing set up so that a single appearance of that line is used in both cases should be possible, something like
```c
#if reason
  if (foo) {
    bar;
  } else
#endif // reason
  baz;
```
(note the indentation - that's how we appear to do it in other locations).

Also the same pattern seems to happen in the other locations below as well.

Reply:
Fixed, thanks!